File: src/measurement/channelComparison/bufferedChannelRange/BCR.cpp
#include "../../../chan.h"
#include <iostream>
#include <chrono>

using namespace std;

int main() {
    int i = 50;

    auto start = std::chrono::high_resolution_clock::now();

    for (int n = 0; n < 500000; n++) {

        Chan<int> bufferedChannel(i);

        for (int m = 0; m < i; m++) {
            bufferedChannel.send(0);
        }

        bufferedChannel.close();

        bufferedChannel.foreach([=](int num) mutable {
            int _ = num;
        });
    }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    cout << "Program took: " << elapsed.count() << "\n";

    return 0;
}

File: src/measurement/channelComparison/bufferedChannelRange/BCR.go
package main

import (
    "log"
    "time"
)

func main() {
    i := 50

    start := time.Now()

    for n := 0; n < 500000; n++ {

        bufferedChannel := make(chan int, i)

        for m := 0; m < i; m++ {
            bufferedChannel <- 0
        }

        close(bufferedChannel)

        for ele := range bufferedChannel {
          _ = ele
        }
    }

    elapsed := time.Since(start)
    log.Printf("Program took %s", elapsed)
}
